va/validation-authority.go

```diff
@@ -606,24 +606,58 @@ func newCAASet(CAAs []*dns.CAA) *CAASet {
 func (va *ValidationAuthorityImpl) getCAASet(ctx context.Context, hostname string) (*CAASet, error) {
 	hostname = strings.TrimRight(hostname, ".")
 	labels := strings.Split(hostname, ".")
+
+	type result struct {
+		Records []*dns.CAA
+		Err     error
+	}
+
+	results := make([]result, len(labels))
+	var wg sync.WaitGroup
+
+	// The lookups are performed in parallel in order to avoid timing out
+	// the RPC call.
+	lookup := func(ctx context.Context, name string, index int) {
+		defer wg.Done()
+
+		CAAs, err := va.DNSResolver.LookupCAA(ctx, name)
```

Comment:
And once the function is that short and has no ifs, move `defer wg.Done()` to the bottom and remove defer.